bl/handler/tournament-handler.go

```diff
@@ -0,0 +1,109 @@
+package handler
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/kimbellG/kerror"
+	ttgrpc "github.com/kimbellG/tournament-bl/handler/grpc"
+	"github.com/kimbellG/tournament-bl/models"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (sh *ServiceHandler) CreateTournament(ctx context.Context, r *ttgrpc.CreateTournamentRequest) (*ttgrpc.CreateTournamentResponse, error) {
+	id, err := sh.tournamentController.Create(tournamentFromProto(r))
+	if err != nil {
+		return &ttgrpc.CreateTournamentResponse{}, kerror.Errorf(err, "controller")
```

Comment:
this is a grpc handler and grpc handler's can't simply return your kerror.Error. Here what you have to do it to map your kerror.Error to grpc status.Code ). For example let's assume that tournament.Controller.Create returned you kerror.Error{Code: InvalidID, Error: errors.New("some error")}. In grpc handler you have to map this Invalid to status.CodeInvalidRequest, and put error as a status message. So the main point is that you have to map kerror.Error to grpc Status.